docs/fileformat-prose.rst

```diff
@@ -0,0 +1,170 @@
+On-disk format
+--------------
+
+.. note::
+   These docs are written for anndata 0.7.
+   Files written before this version may differ in some conventions, but will still be read by newer versions of the library.
+
+AnnData objects are saved on disk to hierarchichal array stores like `HDF5 <https://en.wikipedia.org/wiki/Hierarchical_Data_Format>`_ (via `h5py <http://docs.h5py.org/en/stable/>`_) and `Zarr <https://zarr.readthedocs.io/en/stable/>`_.
+This allows us to have very similar structures in disk and on memory.
+
+In general, `AnnData` objects can hold three kinds of values: (1) dense arrays, (2) sparse arrays, and (3) DataFrames.
+As an example we’ll look into a typical `.h5ad` object that’s been through an analysis.
+This structure should be largely equivalent to Zarr structure, though there are a few minor differences.
+
+.. I’ve started using h5py since I couldn’t figure out a nice way to print attributes from bash.
+
+.. code:: python
+
+   >>> import h5py
+   >>> f = h5py.File("02_processed.h5ad", "r")
+   >>> list(f.keys())
+   ['X', 'layers', 'obs', 'obsm', 'uns', 'var', 'varm'] 
+
+.. .. code:: bash
+
+..    $ h5ls 02_processed.h5ad
+..    X                        Group
+..    layers                   Group
+..    obs                      Group
+..    obsm                     Group
+..    uns                      Group
+..    var                      Group
+..    varm                     Group
+
+Dense arrays
+~~~~~~~~~~~~
+
+Dense arrays have the most simple representation on disk, as they have native equivalents in `HDF5 Datasets <http://docs.h5py.org/en/stable/high/dataset.html#>`_ and `Zarr Arrays <https://zarr.readthedocs.io/en/stable/tutorial.html#creating-an-array>`_.
```

Comment:
I'm having trouble getting this to work, but it might be a caching issue.

Do you think this should look different or just have a different way of linking?

Reply:
A way of linking that survives them changing things. I don’t want our docs to break because they renamed their headers.

Reply:
Wouldn't something like <code>:ref:`h5py:high/dataset`</code> also break if they changed their headers?

Reply:
Well, in this case they’d need to rename the file. Links to `#fragments` easily and often break if you don’t use `:ref:`

Reply:
I'm not quite getting this to work. I'd like to just merge this, and would be happy for you to reformat the links to use inter sphinx (though I'd prefer them to stay inline).

Reply:
Oh, sorry, stuff in `std:doc` has to be referenced with `:doc:`, stuff in `std:label` with `:ref:`